Test/Invoke-MigrationOperation.Tests.ps1

```diff
@@ -0,0 +1,354 @@
+        New-Item -Path $tempPluginsPath -ItemType 'Directory'
+        $pluginPath = Join-Path -Path $PSScriptRoot -ChildPath '..\Rivet\Extras\*-MigrationOperation.ps1' -Resolve
+        Copy-Item -Path $pluginPath -Destination $tempPluginsPath
+        Start-RivetTest -PluginPath $tempPluginsPath
+    }
+    
+    AfterEach {
+        Stop-RivetTest
+        if( (Test-Path -Path $tempPluginsPath -PathType Container) )
+        {
+            Remove-Item -Path $tempPluginsPath -Recurse
+        }
+    }
+    
+    It 'should run plugins' {
+        @'
+    function Push-Migration
+    {
+        Add-Schema 'fubar'
+        Add-Table -SchemaName 'fubar' Foobar -Description 'Test' {
+            BigInt ID -Description 'Test'
+        }
+    }
+    
+    function Pop-Migration
+    {
+        Remove-Table -SchemaName 'fubar' 'Foobar'
+        Remove-Schema 'fubar'
+    }
+    
+'@ | New-TestMigration -Name 'CompleteAdminPlugin'
+    
+        Invoke-RTRivet -Push 'CompleteAdminPlugin'
+    
+        Assert-Column -Name CreateDate -DataType smalldatetime -NotNull -TableName "Foobar" -SchemaName 'fubar'
+        Assert-Column -Name LastUpdated -DataType datetime -NotNull -TableName "Foobar" -SchemaName 'fubar'
+        Assert-Column -Name RowGuid -DataType uniqueidentifier -NotNull -RowGuidCol -TableName "Foobar" -SchemaName 'fubar'
+        Assert-Column -Name SkipBit -DataType bit -TableName "Foobar" -SchemaName 'fubar'
+        Assert-Index -TableName 'Foobar' -ColumnName 'rowguid' -Unique -SchemaName 'fubar'
+        Assert-Trigger -Name 'trFoobar_Activity' -SchemaName 'fubar'
+    }
+    
+    It 'plugins should skip row guid' {
+        @'
+    function Push-Migration
+    {
+        Add-Table Foobar -Description 'Test' {
+            uniqueidentifier guid -RowGuidCol -Description 'Test'
+        }
+    }
+    
+    function Pop-Migration
+    {
+        Remove-Table Foobar
+    }
+'@ | New-TestMigration -Name 'SkipRowGuid'
+    
+        Invoke-RTRivet -Push 'SkipRowGuid'
+    
+        Assert-Column -Name guid -DataType uniqueidentifier -RowGuidCol -TableName "Foobar"
+        (Test-Column -TableName 'Foobar' -Name 'rowguid') | Should Be $false
+    }
+    
+    It 'should reject triggers without not for replication' {
+        $m = @'
+    function Push-Migration
+    {
+        Add-Table Foobar -Description 'Test' {
+            int ID -Identity -Description 'Test'
+        }
+        
+        $trigger = @"
+    ON [dbo].[Foobar]
+    FOR UPDATE
+    AS
+        RETURN
+"@
+    
+        Add-Trigger 'trFoobar_Nothing' -Definition $trigger
+    }
+    
+    function Pop-Migration
+    {
+        Remove-Table 'Foobar'
+    }
+'@ | New-TestMigration -Name 'ValidateMigrations'
+    
+        try
+        {
+            Invoke-RTRivet -Push 'ValidateMigrations' -ErrorAction SilentlyContinue
+            (Test-Trigger -Name 'trFoobar_Nothing') | Should Be $false
+            $Global:Error.Count | Should BeGreaterThan 0

```

Comment:
Maybe this was intentional but the rest of this PR uses the Pester v4 switch syntax, e.g. `Should -BeGreaterThan 0`, but this whole file uses the v3 syntax, e.g. `Should BeGreaterThan 0`